remotespark/livyclientlib/sparkcontroller.py

```diff
@@ -70,7 +71,10 @@ def add_session(self, name, connection_string, skip_if_exists, properties):
 
         session = self.client_factory.create_session(connection_string, properties, "-1", False)
         session.start()
-        livy_client = self.client_factory.build_client(session)
+
+        msg_printer = MessagePrinter.MessagePrinter()
```

Comment:
This doesn't work, this would tie livyclientlib to IPython.  You need to make this object an argument to the method so you can create it elsewhere and mock it.